src/components/rangeslider/draw.js

```diff
@@ -92,8 +95,13 @@ module.exports = function(gd) {
         opts._height = (fullLayout.height - margin.b - margin.t) * opts.thickness;
         opts._offsetShift = Math.floor(opts.borderwidth / 2);
 
-        var x = margin.l + (graphSize.w * domain[0]),
-            y = fullLayout.height - opts._height - margin.b;
+        var x = margin.l + (graphSize.w * domain[0]);
+
+        var y = (
+            margin.t + graphSize.h * (1 - oppDomain[0]) +
+            tickHeight +
+            opts._offsetShift + constants.extraPad
+        );
```

Comment:
The image you've pasted here is blown up (with browser zoom?), so the rounding isn't quite right and consequently there are still some odd 1-pixel effects at the corners. When I look at the rangeslider-crisp-round branch locally it looks *almost* perfect.
![screen shot 2017-02-22 at 10 22 28 pm](https://cloud.githubusercontent.com/assets/2678795/23243514/8144f588-f94d-11e6-8d2e-d7fd2fd1b38e.png)

I realize this is a borderline obsessive ask (which side of the border?) so I did it myself in https://github.com/plotly/plotly.js/commit/29a31d5a0fc7bea0f1b6a7831358ca7a224202e1 and https://github.com/plotly/plotly.js/commit/b49ea77012de7fbb6f80f1af38cd37d389e53433. I just tweaked it a tiny bit to get the symmetry right: both handles the slightest bit more on the dark side. I suppose we could bump the width either up or down 1px so it would be possible to have them each exactly symmetric across the dark/light line, but to my eye anyway symmetry between the two handles is more important. Also included rounding of everything that goes into the grabber position before offsetting, because sometimes we were still getting a non-integer starting point, making the half-pixel offset moot. Kind of annoying but I think now it's really truly guaranteed to be an honest rectangle.

![screen shot 2017-02-22 at 10 32 16 pm](https://cloud.githubusercontent.com/assets/2678795/23243976/52ceb088-f950-11e6-8890-6225172e8302.png)

Seem reasonable?